### PR TITLE
request-server: don't return EOF if there is an unexpected error

### PR DESCRIPTION
**Pull request by drakkan**
io.EOF means [no error](https://github.com/pkg/sftp/blob/master/examples/request-server/main.go#L124) so convert to ErrUnexpectedEOF if we detect a transfer error

### CODE REVIEW

**Review by davecheney — CHANGES_REQUESTED** [review]
Please add a test. Thank you

**Comment by drakkan** [comment]
> Please add a test. Thank you

Done, thank you

File: request-server_test.go [hunk 1]
```diff
@@ -39,6 +40,9 @@ func clientRequestServerPair(t *testing.T) *csPair {
 	skipIfPlan9(t)
 	ready := make(chan bool)
 	os.Remove(sock) // either this or signal handling
+	pair := &csPair{
+		svrResult: make(chan error),
```
Comment (puellanivis) [inline request-server_test.go:44]: Can you make this with a buffer of 1? That way even if we do not read from this channel, there is not a blocked goroutine left hanging?

Comment (drakkan) [reply request-server_test.go]: Done, thank you

**Review by puellanivis — APPROVED** [review]
I cannot find anything else to even nitpick about. :laughing:

**Comment by drakkan** [comment]
> I cannot find anything else to even nitpick about.

Thank you!

**Pull request merged**